modules/collection/desktops/niri.nix

```diff
@@ -0,0 +1,222 @@
+          parameters = {
+            repeat = false;
+          };
+        };
+      };
+    };
+    spawn-at-startup = mkOption {
+      type = listOf (listOf str);
+      default = [];
+      example = literalExpression ''
+        [
+          ["waybar"]
+          ["${pkgs.alacritty}/bin/alacritty" "-e" "fish"]
+        ]
+      '';
+    };
+    extraVariables = mkOption {
+      type = attrsOf str;
+      default = {};
+      example = {
+        DISPLAY = ":0";
+      };
+      description = ''
+        Extra environmental variables to be added to Niri's `enviroment` node.
+        This can be used to override variables set in {option}`enviroment.sessionVariables`.
+        You can therefore set a variable to `null` to force unset it in Niri.
+      '';
+    };
+    configFiles = mkOption {
+      type = listOf path;
+      default = [];
+      example = literalExpression ''
+        [
+          ./input.kdl
+          ./miscellaneous.kdl
+        ]
+      '';
+      description = ''
+        [niri's wiki]: https://github.com/YaLTeR/niri/wiki/Configuration:-Introduction
+
+        A list of files concat together to make up {file}`$HOME/niri/config.kdl`. See a full
+        list of options in [niri's wiki].
+        To add to environment, please use {option}`environment.sessionVariables`.
+
+        You can optionally import certain files using something like this:
+
+        ```nix
+          # lib.flatten takes the elements of lists inside a list and moves them into one list
+          configFiles = lib.flatten [
+            ./config.kdl
+            (lib.optional (config.powersave.enable) ./laptop-config.kdl)
+            (lib.optional (config.programs.firefox.enable) ./firefox-rules.kdl
+          ];
+          ;
+        ```
+
+        You could also add these optional statements in different files, for example:
+
+        ```nix
+        # niri/default.nix
+          configFiles = [./config.kdl];
+
+        # firefox/default.nix
+          rum.desktops.niri.configFiles = lib.optional (config.programs.firefox.enable) ./niri-rules.kdl;
+        ```
+
+        Be warned, however, that some KDL nodes (such as `binds`) cannot have duplicates. However, this
+        should work great for `window-rule`s, `output`s, and other such situations.
+
+        Finally, if you would like to write a string inside an existing Nix file straight into your config,
+        you can do that with this option, too:
+
+        ```nix
+        # waybar.nix
+          rum.desktops.niri.configFiles = [(pkgs.writeText "spawn-waybar" "spawn-at-startup \"waybar\"")];
+        ```
+
+        This will add the short string "spawn-at-startup "waybar"" to your Niri config.
+      '';
+    };
+    extraConfig = mkOption {
+      type = lines;
+      default = "";
+      example = ''
+        screenshot-path "~/Pictures/Screenshots/Screenshot from %Y-%m-%d %H-%M-%S.png"
+
+        output "HDMI-A-1" {
+          off
+        }
+      '';
```

Comment:
I would add a `literalExample` here, and show an example with interpolated Nix values on top of what we have. I think it's important to hint at interpolated variables being used here in a clean manner, as that is often the best way to do it with modules that have a `configFile` option (or cannot be represented by nix values).